fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java

```diff
@@ -146,6 +149,37 @@ public boolean hasPermission(
     return false;
   }
 
+  public boolean canCreate(String resourceType, Object resource) {
+    if (!fiatStatus.isEnabled()) {
+      return true;
+    }
+
+    String username = getUsername(SecurityContextHolder.getContext().getAuthentication());
+
+    try {
+      return AuthenticatedRequest.propagate(
+              () -> {
+                return retryHandler.retry(
+                    "determine whether " + username + " can create resource" + resource,
+                    () -> {
+                      Response response = fiatService.canCreate(username, resourceType, resource);
+                      if (response.getStatus() == HttpServletResponse.SC_OK) {
+                        return true;
+                      } else if (response.getStatus() == HttpServletResponse.SC_NOT_FOUND) {
+                        return false;
+                      } else {
+                        throw new ServletException(
```

Comment:
lets pick something other than ServletException here. I think IntegrationException with setRetryable(true) would be more appropriate (we may want to smarten up retry handler in the future, so there is a preference to use subclasses of SpinnakerException wherever possible)

Reply:
should have spinnaker/kork#401 merged in to make that a litle less painful

Reply:
will also not want to retry on BAD_REQUEST

Reply:
see https://github.com/spinnaker/fiat/pull/485/files#r337840808